### PR TITLE
cba inputs rest api

### PR DESCRIPTION
**Pull request by avpeery**
### Why:
* The PR completes the requirements of adding new cba inputs for new load profiles on the cba page. In addition, some cleaning up of frontend files (some formatting, some clearer names, and introduction of a helper functions file)

### What:
* To do this, it required a new api endpoint that runs the cost benefit analysis runner when called.

### CODE REVIEW

File: webserver/script/CostBenefitAnalysis/UploadToPostgres.py
Comment (derins) [inline webserver/script/CostBenefitAnalysis/UploadToPostgres.py]: Can we remove this?

Comment (avpeery) [reply webserver/script/CostBenefitAnalysis/UploadToPostgres.py]: thanks for the catch - will do, and I'll do a sweep check of print statements to make sure.

File: frontend/src/Components/Helpers/helpers.js
Comment (jongoncalves) [inline frontend/src/Components/Helpers/helpers.js]: If this is ever true then that means the previous conditional threw an exception that isn't being handled. Swap these around.

**Review by jongoncalves — COMMENTED** [review]
Other than the one comment, it looks good to me.

**Comment by avpeery** [comment]
@derins @jongoncalves  changes made - let me know if it's got the approval and I'll squash/merge it in! thanks!

**Review by jongoncalves — APPROVED** [review]

**Comment by avpeery** [comment]
@derins  @jongoncalves , opening this PR again. In addition to the changes above for the cba inputs rest api, this PR now uses the preprocessing - cba tool workflow that wasn't implemented before. There are a lot of files here, provided by Robbie from E3, required for the preprocessing module and cba module to run. Some files for the preprocessing module have been added to .gitignore for the NDA. Please note that while the preprocessing module is running correctly, we are expecting a new CBA Tool soon from Robbie. The current CBA tool here is the original one that only processes the original EV load profiles. The new tool that we are expecting from Robbie will accept the processed csv files that I am currently saving under the "outputs" directory in preprocessing_loadprofiles. In the meantime, this implementation here still runs the same workflow (with a minor supplementation for EV profile csv's), and the time / processing requirements are what should also be expected with the new CBA tool as well. I am going to go through the top of each file that I wrote code on, as there as are many csv files here. Thanks!

File: .gitignore
Comment (avpeery) [inline .gitignore:380]: Added csv data files for preprocessing module to gitignore

File: frontend/src/Components/AlgInputs/AlgInputs.js
Comment (avpeery) [inline frontend/src/Components/AlgInputs/AlgInputs.js:29]: This was reviewed in initial PR - moved function to helper function for better reusability.

File: frontend/src/Components/AlgInputs/AlgInputsCBA.js
Comment (avpeery) [inline frontend/src/Components/AlgInputs/AlgInputsCBA.js:4]: frontend changes in this file from initial PR, since initial approval, not much has changed in this file.

File: frontend/src/Components/Download/Download.js
Comment (avpeery) [inline frontend/src/Components/Download/Download.js:1]: These are changes from applying ESLint for formatting

File: frontend/src/Components/Helpers/helpers.js
Comment (avpeery) [inline frontend/src/Components/Helpers/helpers.js:1]: helper function added for reusability for visualizeResults function

File: frontend/src/Components/Result/ResultCharts.js
Comment (avpeery) [inline frontend/src/Components/Result/ResultCharts.js:9]: This change enabled the correct legend implementation

File: webserver/script/CostBenefitAnalysis/preprocessing_loadprofiles/split_file.py
Comment (avpeery) [inline webserver/script/CostBenefitAnalysis/preprocessing_loadprofiles/split_file.py:1]: This code is originally from Robbie to preprocess the csv output from load forecast profile to then feed to the CBA tool. Some changes here from me.

File: webserver/script/LoadForecasting/LoadForecastingRunner.py
Comment (avpeery) [inline webserver/script/LoadForecasting/LoadForecastingRunner.py:57]: Added this line to save the dataframe as a csv to run through the preprocessing module.

File: webserver/script/views.py
Comment (avpeery) [inline webserver/script/views.py:28]: api endpoint below for the costbenefit analysis runner. Runs the preprocessing module to prepare the load forecast csv profile to run through the CBA tool. Then uploads the CBA tool results to postgres CBA input tables.

File: webserver/script/views.py
Comment (avpeery) [inline webserver/script/views.py:36]: api endpoint below for the costbenefit analysis runner. Runs the preprocessing module to prepare the load forecast csv profile to run through the CBA tool. Then uploads the CBA tool results to postgres CBA input tables.

File: webserver/script/CostBenefitAnalysis/preprocessing_loadprofiles/split_file.py
Comment (derins) [inline webserver/script/CostBenefitAnalysis/preprocessing_loadprofiles/split_file.py]: can we remove these comments?

File: webserver/script/LoadForecasting/LoadForecastingRunner.py
Comment (derins) [reply webserver/script/LoadForecasting/LoadForecastingRunner.py:57]: Path leads to a non-existent directory

**Review by derins — CHANGES_REQUESTED** [review]
In general I would prefer to see some of the unrelated frontend changes in different PRs in the future as this one is already hefty

File: webserver/script/CostBenefitAnalysis/preprocessing_loadprofiles/split_file.py
Comment (avpeery) [reply webserver/script/CostBenefitAnalysis/preprocessing_loadprofiles/split_file.py]: We may possibly implement the weekend version in the future, so I am not comfortable removing the weekend option yet

File: webserver/script/LoadForecasting/LoadForecastingRunner.py
Comment (avpeery) [reply webserver/script/LoadForecasting/LoadForecastingRunner.py:57]: It is in the gitignore. I will fix it so the directory is committed

Comment (derins) [reply webserver/script/LoadForecasting/LoadForecastingRunner.py:57]: Thanks

File: webserver/script/CostBenefitAnalysis/preprocessing_loadprofiles/split_file.py
Comment (derins) [reply webserver/script/CostBenefitAnalysis/preprocessing_loadprofiles/split_file.py]: Alright then

File: webserver/script/LoadForecasting/LoadForecastingRunner.py
Comment (avpeery) [reply webserver/script/LoadForecasting/LoadForecastingRunner.py:57]: fixed - thanks derin!

**Comment by derins** [comment]
I'm okay with merging this after @jongoncalves 's approval, thanks @avpeery

**Comment by avpeery** [comment]
@derins  fyi - just added a new commit merging in your new commits to master

File: frontend/src/Components/AlgInputs/AlgInputsCBA.js
Comment (jongoncalves) [inline frontend/src/Components/AlgInputs/AlgInputsCBA.js]: The `profileName` is saved in the state and referenced as TextField with a `onChange` event already set. You should never have to query the DOM. Just get the value from the state.

As the value in the UI changes, the onChange is triggered and the setProfileName is called and should always have the latest value. Remove all the dom queries and replace with the `this.state.profileName`

File: frontend/src/Components/AlgInputs/AlgInputsLoadControl.js
Comment (jongoncalves) [inline frontend/src/Components/AlgInputs/AlgInputsLoadControl.js]: This is git's merge conflict markers. Please fix the merge conflicts and remove them. The code won't work otherwise.

File: frontend/src/Components/AlgInputs/AlgInputsLoadControl.js
Comment (jongoncalves) [inline frontend/src/Components/AlgInputs/AlgInputsLoadControl.js]: Do we need this comment or can we just remove this block of code.

File: frontend/src/Components/AlgInputs/AlgInputsLoadForecast.js
Comment (jongoncalves) [inline frontend/src/Components/AlgInputs/AlgInputsLoadForecast.js:107]: 0 coerces to false, you can just simplify this to `if(!config_res.data.length)`

File: frontend/src/Components/Helpers/helpers.js
Comment (jongoncalves) [inline frontend/src/Components/Helpers/helpers.js]: same length comment as above

File: frontend/src/Components/Result/ResultCharts.js
Comment (jongoncalves) [inline frontend/src/Components/Result/ResultCharts.js]: same length comment as above

File: webserver/script/CostBenefitAnalysis/python_code/boolean.py
Comment (jongoncalves) [inline webserver/script/CostBenefitAnalysis/python_code/boolean.py:2]: This file is such a foreign idea in python to me.....I'm assuming we didn't write this so I guess it's ok.

File: webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py
Comment (jongoncalves) [inline webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py:32]: Did we write this class?
We need to re-write this and favor using `not dict_object_im_evaluating` instead of `self.dict_object_im_evaluating == {}`.
Also please don't nest the conditionals like this and we need to `return` on all code paths.

File: webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py
Comment (jongoncalves) [inline webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py:164]: Do we know what exactly we should be catching here?

File: webserver/script/views.py
Comment (jongoncalves) [inline webserver/script/views.py:40]: It's been a bit hard to follow, but what is this ModelInstance instantiation doing?

**Review by jongoncalves — COMMENTED** [review]
It seems like this file `building_loads.py` is completely blank/empty. Can we remove it?

File: webserver/script/CostBenefitAnalysis/python_code/boolean.py
Comment (avpeery) [reply webserver/script/CostBenefitAnalysis/python_code/boolean.py:2]: Yes sorry it this isn't clear but inside the python_code directory, the only file from us (well edited file from CMU students) is UploadToPostgres. The python_code directory is a part of the cba_tool provided by Robbie. My apologies for not making that more clear.

File: frontend/src/Components/AlgInputs/AlgInputsLoadForecast.js
Comment (avpeery) [reply frontend/src/Components/AlgInputs/AlgInputsLoadForecast.js:107]: Yes will change thanks!

File: webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py
Comment (avpeery) [reply webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py:164]: I do not. The cba_tool is supposed to be seen as a "black box". I know Robbie is also in the process of writing a new version of the CBA tool as well.

File: webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py
Comment (avpeery) [reply webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py:32]: this is code not written by us. I can get in touch with Robbie and see if it ok to make these changes, but I know we will be receiving a new cba tool relatively soon

File: webserver/script/CostBenefitAnalysis/python_code/boolean.py
Comment (avpeery) [reply webserver/script/CostBenefitAnalysis/python_code/boolean.py:2]: yes python_code directory was not written by us, only file is UploadToPostgres. Again my apologies! Thanks for going through!

File: webserver/script/views.py
Comment (avpeery) [reply webserver/script/views.py:40]: ModelInstance() is how to call the cba_tool to run. It takes in the csv's in EV loads/ load profiles, runs various algorithms to figure out the cost benefit analysis and then populates the results directory with new csv's containing the results. UploadToPostgres then inserts these results to the database.

Comment (avpeery) [reply webserver/script/views.py:40]: I can hop on a quick zoom call tomorrow to do a walk through if needed. I know it's a lot in this PR. Thanks

File: frontend/src/Components/AlgInputs/AlgInputsLoadControl.js
Comment (avpeery) [reply frontend/src/Components/AlgInputs/AlgInputsLoadControl.js]: my apologies. I merged some changes from master recently, and didn't catch this. Thanks.

File: frontend/src/Components/AlgInputs/AlgInputsLoadControl.js
Comment (avpeery) [reply frontend/src/Components/AlgInputs/AlgInputsLoadControl.js]: removed

**Comment by avpeery** [comment]
@jongoncalves I made all the changes requested. I apologize for not stating previously that the files in the python_code directory are a part of the CBA Tool created by Robbie. We are expecting a new version of the CBA tool relatively soon, but for now this version runs as close as possible, and should be just a quick plug in with the new CBA tool when received. I also made another quick change just now of moving the UploadToPostgres.py for the CostBenefitAnalysis out of the python_code directory since that is code from us and not a part of the CBA tool, for better maintainability/understanding.

**Review by jongoncalves — APPROVED** [review]

**Review by derins — APPROVED** [review]

**Pull request merged**
